spec/system/publishers/publishers_can_view_statistics_spec.rb

```diff
@@ -0,0 +1,105 @@
+require "rails_helper"
+
+RSpec.describe "Publishers can view statistics" do
+  let(:school) { create(:school) }
+
+  let(:publisher) { create(:publisher, publisher_preferences: build_list(:publisher_preference, 1, organisation: school)) }
+
+  before do
+    create(:vacancy, organisations: [school], publisher: publisher)
+    login_publisher(publisher: publisher, organisation: school)
+  end
+
+  after { logout }
+
+  describe "vacancy stats page" do
+    before do
+      # This old vacancy shouldn't show up in the visible stats as its too old
+      create(:vacancy, publisher: publisher, organisations: [school], publish_on: 1.year.ago,
+                       vacancy_analytics: build(:vacancy_analytics,
+                                                referrer_counts: { "Google" => 6, "Magic" => 8 }))
+      create(:vacancy, publisher: publisher, organisations: [school],
+                       vacancy_analytics: build(:vacancy_analytics,
+                                                referrer_counts: { "Google" => 6, "Magic" => 8 }))
+      create(:vacancy, publisher: publisher, organisations: [school],
+                       vacancy_analytics: build(:vacancy_analytics,
+                                                referrer_counts: { "Google" => 4, "Yahoo" => 12, "LinkedIn" => 13 }))
+      create(:vacancy, publisher: publisher, organisations: [school],
+                       vacancy_analytics: build(:vacancy_analytics,
+                                                referrer_counts: { "Google" => 14, "Indeed" => 5, "LinkedIn" => 17 }))
+    end
+
+    it "adds up the stats for each referrer" do
+      visit publishers_current_year_statistics_path
+
+      find_by_id("accessible").click
+
+      within("#analytics") do
+        within(".govuk-summary-list__row:nth-child(1)") do
+          expect(page).to have_content("LinkedIn")
+          expect(page).to have_content("30")
+        end
+        within(".govuk-summary-list__row:nth-child(2)") do
+          expect(page).to have_content("Google")
+          expect(page).to have_content("24")
+        end
+      end
+    end
+  end
+
+  describe "equal opportunities page" do
+    before do
```

Comment:
It seems to be a lot of test setup info in the `equal_opportunities_report` instances that doesn't seem to be used for any assertion. 

Don't know if it was forgotten or just want to keep the assertions lean. In the second case, I guess we could just not set up the data we don't assert against?